lib/esp-scanning-lib.robot

```diff
@@ -0,0 +1,173 @@
+*** Settings ***
+Documentation       Collection of keywords related to UEFI Secure Boot
+
+Resource            ../keywords.robot
+
+
+*** Variables ***
+${SHIMX64_URL}=     https://cloud.3mdeb.com/index.php/s/TipPHgsCj6MCZHi/download/shimx64.efi
+${GRUBX64_URL}=     https://cloud.3mdeb.com/index.php/s/e2JjYANjMsgyXtS/download/grubx64.efi
+${TINYCORE_URL}=    https://distro.ibiblio.org/tinycorelinux/14.x/x86/release/CorePlus-current.iso
+${DTS_URL}=         https://dl.3mdeb.com/open-source-firmware/DTS/v1.2.8/dts-base-image-v1.2.8.iso
+
+
+*** Keywords ***
+Prepare EFI Partition With System Files
+    [Documentation]    Prepares a bootable EFI partition with OS files
+    ...    by mounting a hard drive in Qemu, or creating files on the DUT
+    ...    via PiKVM
+
+    IF    "${MANUFACTURER}" == "QEMU"
+        Add HDD To Qemu    img_name=image.img
+    ELSE
+        IF    "${DUT_CONNECTION_METHOD}" == "pikvm"
+            Boot System Or From Connected Disk    ubuntu
+            Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+            Login To Linux
+            Switch To Root User
+            Remove All Supported Systems From Efi
+            Populate /EFI With System Folders
```

Comment:
It makes me think if we really need all of these keywords...
If we already have boot image with desired files, we could just mound it and copy everything over? Let's at least create an issue on that as a potential improvement

Reply:
Proposed a slimmer version in [5be206d](https://github.com/Dasharo/open-source-firmware-validation/pull/88/commits/5be206d1f694ca6ceb90fd8fd514b512e92ee9c0)

Reply:
Thanks